ivsonv/cofrinho-bo
# [0001] - Configurations setup coding, lint, conventions

# Why it was done
These changes were made to improve code consistency and standardization, and to facilitate development with recommended tools and configurations.

## What was done
* config `linting` with `prettier` to maintain code `consistency`
* config `formatOnSave = true` in `.vscode` file `setting.json` (apply lint OnSave)
* config `absolute imports` to improve import organization in files.
* add `.editorconfig` to standardize code formatting.
* add `.nvmrc` to specify the Node.js version to be used.
* add `.vscode` plugin recommendations in `extensions.json` (prettier).
* add default pull request template in `.github.`

## How to test
* verify linting coding: npm run `lint:check`
<img width="1389" alt="image" src="https://github.com/user-attachments/assets/885e2448-1697-4c05-b49a-40bf5fe804ed">

* apply `changes` in coding: npm run `lint:fix`
<img width="1184" alt="image" src="https://github.com/user-attachments/assets/257c43ed-3ac2-4cb6-ae2e-bf432c6f8c85">


## Checklist:

- [x] I performed a self-review of my own code
- [x] Code passed linting
- [x] Testing locally
